File: S5/S5/PF/0 introduction/somme.hs
module Somme where

somme n = sum [1..n]
-- [..] est une notation pour la fonction enumFromTo
somme' n = sum (enumFromTo 1 n)

sommePairs n = sum [2,4..n]
-- [, ..] est une notation pour la fonction enumFromThenTo
sommePairs' n = sum (enumFromThenTo 2 4 n)

sommeCarres n = sum (map (^2) [1..n])

File: S5/PF/pf/0 introduction/somme.c
int somme(int n) {
    int i;
    int somme = 0;

    for (i = 1; i <= n; i++) {
        somme += i;
    }

    return somme;
}

int sommePairs(int n) {
    int i;
    int somme = 0;

    for (i = 2; i <= n; i+=2) {
        somme += i;
    }

    return somme;
}
